doc/reference/reference_lua/checks.rst

```diff
@@ -0,0 +1,516 @@
+
+        *   -   ``checks('tuple')``
+            -   Check whether the specified value is a :ref:`tuple <box_tuple-new>`
+            -   :ref:`checkers.tuple(value) <checks-checkers-tuple>`
+
+        *   -   ``checks('uint64')``
+            -   Check whether the specified value is a ``uint64`` value
+            -   :ref:`checkers.uint64(value) <checks-checkers-uint64>`
+
+        *   -   ``checks('uuid')``
+            -   Check whether the specified value is :ref:`uuid_object <uuid-module>`
+            -   :ref:`checkers.uuid(value) <checks-checkers-uuid>`
+
+        *   -   ``checks('uuid_bin')``
+            -   Check whether the specified value is :ref:`uuid <uuid-module>` represented by a 16-byte binary string
+            -   :ref:`checkers.uuid_bin(value) <checks-checkers-uuid_bin>`
+
+        *   -   ``checks('uuid_str')``
+            -   Check whether the specified value is :ref:`uuid <uuid-module>` represented by a 36-byte hexadecimal string
+            -   :ref:`checkers.uuid_str(value) <checks-checkers-uuid_str>`
+
+
+
+.. _checks_custom_function:
+
+Custom function
+***************
+
+A string type qualifier can accept the name of a custom function that performs arbitrary validations.
+To achieve this, the function should be added to the :ref:`checkers <checks-checkers>` table and should return ``true`` if the value is valid.
+
+The example below shows how to use the ``positive`` function to check that an argument value is a positive number.
+
+..  literalinclude:: /code_snippets/test/checks/checks_function_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+.. _checks_metatable_type:
+
+Metatable type
+**************
+
+A string qualifier can accept a value stored in the ``__type`` field of the argument `metatable <https://www.lua.org/pil/13.html>`_.
+
+..  literalinclude:: /code_snippets/test/checks/checks_metatable_type_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+.. _checks_union_type:
+
+Union types
+~~~~~~~~~~~
+
+To allow an argument to accept several types (a union type), concatenate type names with a pipe (``|``).
+In the example below, the argument can be both a ``number`` and ``string`` value.
+
+..  literalinclude:: /code_snippets/test/checks/checks_type_combination_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+
+.. _checks_optional_type:
+
+Optional types
+~~~~~~~~~~~~~~
+
+To make any of the supported types optional, prefix its name with a question mark (``?``).
+In the example below, the ``name`` argument is optional.
+This means that the ``greet`` function can accept ``string`` and ``nil`` values.
+
+..  literalinclude:: /code_snippets/test/checks/checks_string_optional_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+As for a specific type, you can make a :ref:`union type <checks_union_type>` value optional: ``?number|string``.
+
+
+.. _checks_any_type:
+
+Turning off argument checking
```

Comment:
Turn off > Skip?

Reply:
Not sure am I right or not but to me `skip` has a slightly different meaning and suits better for cases when we have some kind of a loop. Here, we turn off checking of the specified argument.

Reply:
I see it this way: 
- by default, no args are checked - checking is _turned off_.
- _turn off_ checking_ means return the default (no check) after _turning on_ checking.
- but this text tells about the case when nobody turned on checking for these args :) 

"skip" came to me because I see args as a sequence. So I read a `check` call with a `?` like this: check `arg1`, check `arg2`, skip `arg3`, check `arg4`.

Reply:
Now I got this :) Will update